### ISSUE TITLE
Strip emojis from reportbacks 

### ISSUE THREAD

**Issue by ngjo** [DoSomethingArchive/legacy-website#7014]
#### User Story

As a member, I should know that I can't submit an emoji in the RB caption or Why cuz it'll 404.
#### Additional Information (optional)

If you submit an emoji it will break the campaign page -- we're getting zendesk tickets from people who are saying after they try to submit a RB, the page 404s and we think it could be someone submitting emojis.

Easier (and interim) fix to #6167 !
#### Why This Matters

So the site doesn't break and the user knows that they can't put in an emoji. 

Here's the text to add in the two places:

![screen shot 2016-09-06 at 5 23 47 pm](https://cloud.githubusercontent.com/assets/2658867/18316544/07877574-74e9-11e6-9db9-04f6b5b5863c.png)

![screen shot 2016-09-06 at 5 23 56 pm](https://cloud.githubusercontent.com/assets/2658867/18316545/0907c084-74e9-11e6-8761-a83924255e9a.png)

cc @hbghidey

**Comment by angaither** [comment]
we might want to hold off on this, and wait till we are sending rbs to rogue.

**Comment by ngjo** [comment]
@angaither with the hope that rogue will accept the emojis? if so, would that mean that ppl could send in emojis?!?!

cc @hbghidey

**Comment by angaither** [comment]
@ngjo not the hope that rogue will accept emojis, the DEMAND that it does ;)

**Comment by ngjo** [comment]
😂😂😂😂😂😂😂😂😂😂😂

i would say there must be a good gif for this, but im too busy lol'ing about that. YESSSSSS!!!! DEMANDS!!!!

Thanks @angaither !

**Comment by ngjo** [comment]
@angaither know that Katie just worked on this for Rogue, but since it won't be turned on for at least some time, would it be possible to add this text in for the interim? 

We've been getting a lot more RBs as everyone ramps up efforts to get ppl reportingback more, buttttt that has meant more people getting the 404 experience.

Let me know!

**Comment by angaither** [comment]
@ngjo sure we can put this into the next sprint

**Comment by ngjo** [comment]
woohoo @angaither thanks!

**Comment by ngjo** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [As a developer, I want to add text to the Why and Caption to indicate that we don't accept emojis at this moment so they don't do it and break the system.](https://trello.com/c/dVPUbH7W/82-as-a-developer-i-want-to-add-text-to-the-why-and-caption-to-indicate-that-we-don-t-accept-emojis-at-this-moment-so-they-don-t-do)

**Comment by DFurnes** [comment]
@ngjo @angaither Adding in a "don't use emoji" help text seems a bit hostile... would it make more sense to strip out the emoji in a form alter hook so they don't cause the database error?

``` php
iconv('UTF-8', 'ASCII//IGNORE//TRANSLIT', 'emoji 😅');  // "emoji "
```

**Comment by ngjo** [comment]
@DFurnes hahaha, I guess that is a little aggressive! If we go the route you suggested, would the RB show up to be reviewable by the admin?

**Comment by DFurnes** [comment]
That will strip the emojis (and any other non-ASCII characters) from the string, so if we added that as a form alter hook Drupal would never attempt to save them to the database and cause things to blow up. That should mean the reportback would then be reviewable by the admin.

Sidnote: Ideally the `//TRANSLIT` option would use fallbacks for characters with passable ASCII equivalents (like `é` to `e`) but that doesn't seem to be reliably working when I tested it now. Switching the option order to `ASCII//TRANSLIT//IGNORE` _does_ translate, but then it inserts `?` for non-translatable chars. 🤔

**Comment by ngjo** [comment]
Cool -- thanks for the clarification @DFurnes ! I'll let @angaither decide on this one as I know that Rogue will be able to accept emojis...so whatever's teh easiest! If we got hte route of additional text, I can sync w/ Ben to maybe have something less aggressive sounding.

**Comment by DFurnes** [comment]
Hah sorry, "aggressive" was probably aggressive terminology on my part! 😉 I just feel like we should try to work around any backend quirks if we can, rather than foisting that onto users. Definitely no need to rephrase if we decide to continue with this solution.

**Comment by angaither** [comment]
@DFurnes yeah the ultimate goal here is to just allow reportbacks to have emoji in rogue
and this is a silly temp solution till rogue is accepting reportbacks.

**Comment by itsjoekent** [comment]
Had some problems getting this running, looking at the notes from the community `iconv` has quite a few bugs http://php.net/manual/en/function.iconv.php#usernotes

I'm gonna try use to some regex functions I found on SO instead, temp solution right? 😅

**Comment by ngjo** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Users hitting 404 page -- Emoji Fix](https://trello.com/c/vfFRBOiT/1602-users-hitting-404-page-emoji-fix)

**Comment by aaronschachter** [comment]
#7115 doesn't seem to be stripping emoji when posting to the `/campaigns/:id/reportback` that Gambit and the mobile apps use. We added [emoji-strip](https://www.npmjs.com/package/emoji-strip) to strip emoji's from Gambit submissions as a hotfix.

cc @mirie

**Comment by itsjoekent** [comment]
@aaronschachter I'm guessing some new emoji;s have been added to the ecosystem since we merged this PR. Thanks for hot-fixing it on your end

**Comment by aaronschachter** [comment]
FYI - Mobile Commons just released emoji support last week, so Gambit historically was never sending raw emoji until [recently](https://github.com/DoSomething/gambit/issues/907). Before the emoji release, Mobile Commons stripped them on their end when posting to Gambit.
